knowledge-base/inputs-open-programmatically.md

```diff
@@ -0,0 +1,203 @@
+    <TelerikButton OnClick="@OpenComboBox">Open ComboBox</TelerikButton>
+    <TelerikButton OnClick="@OpenDropDownList">Open DropDownList</TelerikButton>
+    <TelerikButton OnClick="@OpenMultiSelect">Open MultiSelect</TelerikButton>
+</div>
+
+<br />
+<br />
+AutoComplete:
+<TelerikAutoComplete @ref="@AutoCompleteRef"
+                     Id="AC1"
+                     Data="@ValueCollection"
+                     ValueField="@nameof(Product.Name)"
+                     @bind-Value="@StringValue" />
+<br />
+<br />
+ComboBox:
+<TelerikComboBox @ref="@ComboBoxRef"
+                 Id="CB1"
+                 Data="@ValueCollection"
+                 TextField="@nameof(Product.Name)"
+                 ValueField="@nameof(Product.ID)"
+                 @bind-Value="@IntValue" />
+
+<br />
+<br />
+DropDownList:
+<TelerikDropDownList @ref="@DropDownListRef"
+                     Id="DDL1"
+                     Data="@ValueCollection"
+                     TextField="@nameof(Product.Name)"
+                     ValueField="@nameof(Product.ID)"
+                     @bind-Value="@IntValue" />
+
+<br />
+<br />
+MultiSelect:
+<TelerikMultiSelect @ref="@MultiSelectRef"
+                    Id="MS1"
+                    Data="@ValueCollection"
+                    TextField="@nameof(Product.Name)"
+                    ValueField="@nameof(Product.ID)"
+                    @bind-Value="@MultiValues"
+                    Width="350px" AutoClose="false">
+</TelerikMultiSelect>
+
+@code {
+    List<Product> ValueCollection { get; set; } = new();
+    List<int> MultiValues { get; set; } = new();
+    int IntValue { get; set; }
+    string StringValue { get; set; }
+
+    TelerikMultiSelect<Product, int> MultiSelectRef { get; set; }
+    TelerikAutoComplete<Product> AutoCompleteRef { get; set; }
+    TelerikComboBox<Product, int> ComboBoxRef { get; set; }
+    TelerikDropDownList<Product, int> DropDownListRef { get; set; }
+
+    async Task OpenAutoComplete()
+    {
+        await AutoCompleteRef.FocusAsync();
+    }
+
+    async Task OpenComboBox()
+    {
+        await ComboBoxRef.FocusAsync();
+    }
+
+    async Task OpenDropDownList()
+    {
+        await DropDownListRef.FocusAsync();
+    }
+
+    async Task OpenMultiSelect()
+    {
+        await MultiSelectRef.FocusAsync();
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        await base.OnAfterRenderAsync(firstRender);
+
+        if (firstRender)
+        {
+            // open on tab, click, FocusAsync:
+            await js.InvokeVoidAsync("attachFocusHandler", AutoCompleteRef.Id, ".k-autocomplete");
+            await js.InvokeVoidAsync("attachFocusHandler", ComboBoxRef.Id, ".k-combobox");
+
+            // open on tab, FocusAsync:
+            await js.InvokeVoidAsync("attachFocusHandler", DropDownListRef.Id);
+            await js.InvokeVoidAsync("attachFocusHandler", MultiSelectRef.Id, ".k-multiselect");
+
+            await OpenMultiSelect();
```

Comment:
```
await OpenMultiSelect();
```
Do we want to open the Multiselect OnInitialized? If not, we can remove that

Reply:
Yes, this was one of the tickets that inspired this KB article.